inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/utils/ImageUtil.kt

```diff
@@ -1,19 +1,14 @@
 package com.rakuten.tech.mobile.inappmessaging.runtime.utils
 
-import android.graphics.Bitmap
+import android.content.Context
+import com.squareup.picasso.Callback
 import com.squareup.picasso.Picasso
-import timber.log.Timber
-import java.lang.Exception
 
 internal object ImageUtil {
-    private const val TAG = "IAM_ImageUtil"
 
     @SuppressWarnings("TooGenericExceptionCaught")
-    fun fetchBitmap(imageUrl: String): Bitmap? =
-        try {
-            Picasso.get().load(imageUrl).priority(Picasso.Priority.HIGH).get()
-        } catch (e: Exception) {
-            Timber.tag(TAG).d(e, "Error on loading image $imageUrl")
-            null
-        }
+    fun fetchImage(imageUrl: String, callback: Callback, context: Context, picasso: Picasso? = null) {
+        (picasso ?: Picasso.get()).load(imageUrl).priority(Picasso.Priority.HIGH)
+            .resize(ViewUtil.getDisplayWidth(context), 0).fetch(callback)
```

Comment:
Does using height = 0 keep the image ratio?

Reply:
yes, the height will be adjusted based on the width